src/geojson/geojson.js

```diff
@@ -0,0 +1,494 @@
+    geometrycollection = parser.parseFromString(geometrycollection, "text/html");
+
+    let feature = "<map-feature><map-featurecaption></map-featurecaption><map-geometry></map-geometry><map-properties></map-properties></map-feature>";
+    feature = parser.parseFromString(feature, "text/html");
+
+    // Template to add coordinates to Geometries
+    let coords = "<map-coordinates></map-coordinates>";
+    coords = parser.parseFromString(coords, "text/html");
+    
+    //console.log(layer);
+    if (jsonType === "FEATURECOLLECTION") {
+        
+        // Setting bbox if it exists
+        if (json.bbox) {
+            layer.querySelector("layer-").insertAdjacentHTML("afterbegin", "<map-meta name='extent' content='top-left-longitude=" + json.bbox[0] + ", top-left-latitude=" + json.bbox[1] + ", bottom-right-longitude=" + json.bbox[2] + ",bottom-right-latitude=" + json.bbox[3] + "'></map-meta>");
+        }
+
+        let features = json.features;
+        //console.log("Features length - " + features.length);
+        for (let l=0;l<features.length;l++) {
+            geojson2mapml(features[l], options, layer);
+        }
+    } else if (jsonType === "FEATURE") {
+
+        let clone_feature = feature.cloneNode(true);
+        let curr_feature = clone_feature.querySelector('map-feature');
+        
+        // Setting bbox if it exists
+        if (json.bbox) {
+            layer.querySelector("layer-").insertAdjacentHTML("afterbegin", "<map-meta name='extent' content='top-left-longitude=" + json.bbox[0] + ", top-left-latitude=" + json.bbox[1] + ", bottom-right-longitude=" + json.bbox[2] + ",bottom-right-latitude=" + json.bbox[3] + "'></map-meta>");
+        }
+
+        // Setting featurecaption
+        let featureCaption = layer.querySelector("layer-").getAttribute('label');
+        if (typeof options.caption === "function") {
+            featureCaption = options.caption(json);
+        } else if (typeof options.caption === "string") {
+            featureCaption = json.properties[options.caption];
+        }
+        curr_feature.querySelector("map-featurecaption").innerHTML = featureCaption;
+
+        // Setting Properties
+        let p;
+        // if properties function is passed
+        if (typeof options.properties === "function") {
+            p = options.properties(json.properties);
+        } else if (typeof options.properties === "string") { // if properties string is passed
+            curr_feature.querySelector('map-properties').insertAdjacentHTML("beforeend", options.properties);
+            p = false;
+        } else if (options.properties instanceof HTMLElement) { // if an HTMLElement is passed - NOT TESTED
+            p = options.properties;
+        } else { // If no properties function, string or HTMLElement is passed
+            p = properties2Table(json.properties);
+        }
+        
+        if (p) {
+            curr_feature.querySelector('map-properties').appendChild(p);

```

Comment:
If my function returns a string I get an error here.  I think it might be better silently ignore non-node returns.  WDYT?  Maybe it's good feedback to the user, not sure.

Reply:
what do we set the properties to when a non-node is returned, I presume we leave it blank. I think it would be good feedback to the user, perhaps I can change the error it throws to guide the user? or perhaps a silent warning in the console, while leaving the properties blank?

Reply:
If properties function now returns a non HTMLElement, it produces a silent error in the console, and leaves the properties empty.